osu.Game/Overlays/Settings/Sections/SkinSection.cs

```diff
@@ -135,6 +133,24 @@ private void itemRemoved(ValueChangedEvent<WeakReference<SkinInfo>> weakItem)
                 Schedule(() => skinDropdown.Items = skinDropdown.Items.Where(i => i.ID != item.ID).ToArray());
         }
 
+        private List<SkinInfo> sortList(List<SkinInfo> skinsList)
+        {
+            skinsList.Sort((a, b) => String.Compare(a.Name, b.Name, StringComparison.Ordinal));
+            for (int i = 0; i < skinsList.Count; i++)
+            {
+                // insert lazer built-in skins before user skins
+                if (skinsList[i].ID <= 0) {
+                    var itemToMove = skinsList[i];
+                    skinsList.RemoveAt(i);
+                    skinsList.Insert(0, itemToMove);
+                }
+            }
+            skinsList.RemoveAll(s => s.ID == SkinInfo.RANDOM_SKIN);
+            skinsList.Insert(0, random_skin_info);
```

Comment:
Several issues with this:

* This needlessly removes three items and reinserts them back to the start again, incurring unnecessary six linear operations on the entire list, while it should only be sorting the section of the list that this cares about (so user skins). `List<T>.Sort()` has an overload that takes an index and count.
* Due to the above this also wrongly pushes the `Random Skin` option to the top, while it intentionally used to be third in the order.